source/common/grpc/google_grpc_creds_impl.cc

```diff
@@ -4,13 +4,83 @@
 #include "envoy/grpc/google_grpc_creds.h"
 #include "envoy/registry/registry.h"
 
+#include "common/config/datasource.h"
+
 namespace Envoy {
 namespace Grpc {
 
-/**
- * TODO: Create GoogleGrpcCredentialsFactory for each built-in credential type defined in Google
- * gRPC.
- */
+std::shared_ptr<grpc::ChannelCredentials>
+defaultSslChannelCredentials(const envoy::api::v2::core::GrpcService& grpc_service_config,
+                             bool allow_insecure) {
+  const auto& google_grpc = grpc_service_config.google_grpc();
+  std::shared_ptr<grpc::ChannelCredentials> creds = nullptr;
```

Comment:
bring L16-21 to the bottom of this function then you don't need variable `creds`.